conans/cli/api/subapi/install.py

```diff
@@ -89,17 +100,29 @@ def conan_full_deploy(conanfile, output_folder):
     import os
     import shutil
 
-    conanfile.output.info(f"Conan built-in full deployer to {output_folder}")
-    for r, d in conanfile.dependencies.items():
-        folder_name = os.path.join(d.context, d.ref.name, str(d.ref.version))
-        build_type = str(d.info.settings.build_type)
-        arch = str(d.info.settings.arch)
+    root_node = graph.root
+    root_conanfile = root_node.conanfile
+
+    root_conanfile.output.info(f"Conan built-in full deployer to {output_folder}")
+    for node in graph.nodes:
```

Comment:
We cannot expect the user to iterate the graph nodes to do the deployers, we need an interface through conanfile.dependencies and skip that logic of virtual, consumers, binary skip and editables.